File: php/features.php
<?php
declare(strict_types=1);

// MultifonClient SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MultifonClientFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MultifonClientBaseFeature();
            case "test":
                return new MultifonClientTestFeature();
            default:
                return new MultifonClientBaseFeature();
        }
    }
}

File: py/multifonclient_sdk/features.py
# MultifonClient SDK feature factory

from multifonclient_sdk.feature.base_feature import MultifonClientBaseFeature
from multifonclient_sdk.feature.test_feature import MultifonClientTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MultifonClientBaseFeature(),
        "test": lambda: MultifonClientTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
